libp2p-floodsub/src/topic.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2018 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use bs58;
+use protobuf::Message;
+use rpc_proto;
+
+/// Represents the hash of a topic.
+///
+/// Instead of a using the topic as a whole, the API of floodsub uses a hash of the topic. You only
+/// have to build the hash once, then use it everywhere.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct TopicHash {
+    hash: String,
```

Comment:
Why is the hash a string and not a fixed-size buffer of `u8`s? This kinda obfuscates the intent.

Reply:
The protocol definition is like that.